peps/pep-0705.rst

```diff
@@ -310,235 +217,260 @@ This is consistent with the behavior introduced in :pep:`655`.
 
-    c: C = { "x": 1, "y": 2 }
-    d: D = { "x": 3, "y": "foo" }
-    update_b(c, d)  # c is no longer a C at runtime
+    class A(TypedDict):
+      v: int | str
+      x: NotRequired[A]
 
-Both mypy and pyright currectly permit this usage, however, as the only viable alternative has been to prevent calling ``update`` at all.
+    class B(TypedDict):
+      v: int
+      x: NotRequired[C]
 
-With the addition of ``other_keys``, it becomes possible to more accurately type the update method:
+    class C(TypedDict):
+      v: str
+      x: NotRequired[B]
 
-* Declare a new read-only TypedDict type that does not allow other keys
-* Copy all non-read-only entries to it
-* Make all entries read-only and non-required
-* Union this with an iterable of matching key-value pairs
+    b: B = { "v": 1, "x": { "v": "two", "x": { "v": 3 } } }
+    a: A = deepcopy(b)  # OK
 
-For instance::
+    class D(TypedDict):
+      v: int
+      x: NotRequired[E]
 
-    class Example(TypedDict):
-        a: int
-        b: NotRequired[str]
-        c: ReadOnly[int]
+    class E(TypedDict):
+      v: float
+      x: NotRequired[D]
 
-    class ExampleUpdateDict(TypedDict, total=False, readonly=True, other_keys=Never):
-        a: int
-        b: str
-        # c is not present as it is read-only in Example
+    d: D = { "v": 1, "x": { "v": 1.1 } }
+    a: A = deepcopy(d)  # Type check error: "v" is of type float in E
 
-    ExampleUpdateEntry = tuple[Literal["a"], int] | tuple[Literal["b"], str]
-    ExampleUpdate = ExampleUpdateDict | Iterable[ExampleUpdateEntry]
+* Type checkers may choose to reject deep copying between complex recursive types for algorithmic complexity reasons. However, the error should be explicit that this is a type checker limitation, and not due to an unsound assignment in the user code.
 
-Type checkers should permit any type compatible with this TypedDict to be passed into the update operation. As with :pep:`589`, they may choose to continue permitting TypedDict types that allow other keys as well, to avoid generating false positives.
 
 Keyword argument typing
 -----------------------
 
-:pep:`692` introduced ``Unpack`` to annotate ``**kwargs`` with a ``TypedDict``. Marking one or more of the entries of a ``TypedDict`` used in this way as read-only will have no effect on the type signature of the method, since all keyword arguments are read-only by design in Python. However, it *will* prevent the entry from being modified in the body of the function::
+:pep:`692` introduced ``Unpack`` to annotate ``**kwargs`` with a ``TypedDict``. Marking one or more of the items of a ``TypedDict`` used in this way as read-only will have no effect on the type signature of the method, since all keyword arguments are read-only by design in Python. However, it *will* prevent the item from being modified in the body of the function::
```

Comment:
```suggestion
:pep:`692` introduced ``Unpack`` to annotate ``**kwargs`` with a ``TypedDict``. Marking one or more of the items of a ``TypedDict`` used in this way as read-only will have no effect on the type signature of the method. However, it *will* prevent the item from being modified in the body of the function::
```
I realize this isn't new text, but the statement doesn't make much sense to me: what does it even mean for a keyword argument to be read-only?